fdbserver/DDShardTracker.actor.cpp

```diff
@@ -75,7 +75,7 @@ ACTOR Future<Void> updateMaxShardSize(Reference<AsyncVar<int64_t>> dbSizeEstimat
 }
 
 struct DataDistributionTracker : public IDDShardTracker {
-	Database cx;
+	std::shared_ptr<IDDTxnProcessor> db;
```

Comment:
Is there any criteria we use to determine whether `Reference` or `shared_ptr` should be used? Mostly I have chosen `Reference` for consistency with the rest of the codebase, but I haven't really given this too much thought.

Reply:
I don’t have strong proof of which one is better. I prefer `std::shared_ptr` because it's more convenient without extra derived class code or `addref` methods.

Reply:
In that case, we should probably seek some consensus on what standard we want to use in our codebase before introducing mixed usage of `shared_ptr` and `Reference`. I don't have any particularly concerns with using `shared_ptr`, but I think it's worth clarifying whether there is an expectation to use our own `Reference` type.

Reply:
After discussion in slack channel, in the future we will always use `Reference` unless we want `weak_ptr`.